bundles/org.eclipse.jface.text/src/org/eclipse/jface/text/contentassist/ContentAssistant.java

```diff
@@ -1580,8 +1579,11 @@ protected void install() {
 		fInternalListener= new InternalListener();
 
 		AdditionalInfoController controller= null;
-		if (fInformationControlCreator != null)
-			controller= new AdditionalInfoController(fInformationControlCreator, OpenStrategy.getPostSelectionDelay());
+		if (fInformationControlCreator != null) {
+			/** delay until each additional information is shown **/
```

Comment:
I think it's better to keep te OpenStrategy.getPostSelection() delay here (and probably improve its value in another PR to match the case of repeated hits on arrow keys)

Reply:
> Therefore i prefer to touch only usecase by usecase.

The thing here is that you're not touching usecase by usecase, but you're touching all use-cases at once. Also, if a delay make sense for some use-cases, it's likely to make sense for some other.

Reply:
enlighten me; what is any other usecase here then showing the additional completion window?  This value was only used to show the wrong (outdated) information for a fixed 0.5 sec and if the computation took too long to never show it. My guess is you still have not tried out what this value does. That was more a bug then a feature.

Reply:
> what is any other usecase here then showing the additional completion window?

Unless we're talking about different thing, any IContentAssistProcessor can return an IContentAssistProposal that returns a `getAdditionalProposalInfo` (which may be long to compute) and that will show an additional completion window (which may be long to render) in order to show completion details. CDT, LSP4E-based projects such as Wild Web Developer or ShellWax or m2e... and many others are example.

>  This value was only used to show the wrong (outdated) information for a fixed 0.5 sec 

I suspect the value here is used to prevent from extra computation and flickering with new information if users performs other actions in a short time. Retaining an outdated value/popup for some ms is more comfortable than "blinking" popup.
Again, the root cause of the whole discussion is that 0.5s is too much and doesn't fit the reality of keyboard navigation. Just moving this value to ~0.3s (in `OpenStrategy`) would make a big difference without challenging all the existing performance or UX optimizations.

Reply:
Just that there is no flickering nor computational saving nor any blinking popup related to this value. The popup open if and only if you press the accelerator and closes when you escape or commit.
That we additional reduce OpenStrategy for other usecase is still an option. Even if rendering takes long it does so after the delay. If you  press keys faster then the rendering (which i still doubt or would qualify you for the tetris world record) the rendering is skipped anyway by the existing double check that the proposal is still the current. This state machine already has a build in throttler. just hard to see because it is a bit overcomplicated